verifier/verifier.go

```diff
@@ -518,6 +545,67 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r, err := revocation.New(client)
+	if err != nil {
+		logger.Debug("no HTTP client provided for revocation checking")
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  errors.New("unable to check revocation status, an HTTP client must be provided"),
+		}
+	}
+
+	authenticSigningTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	// TODO use authenticSigningTime from signerInfo
+	// https://github.com/notaryproject/notation-core-go/issues/38
+	revocationResult := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, authenticSigningTime)
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	// Log all non-nil errors as debug
+	for i, certResult := range revocationResult {
+		for _, err := range certResult.ServerResults {
+			if err != nil {
+				logger.Debugf("error for certificate #%d in chain: %v", (i + 1), err)
+			}
+		}
+	}
+
+	currResult := revocation.OK
+	for _, certResult := range revocationResult {
+		switch certResult.Result {
+		case revocation.OK:
+			continue
+		case revocation.Revoked:
+			currResult = revocation.Revoked
+			// will break out of loop after this
+		default:
+			// revocation.Unknown
+			currResult = revocation.Unknown
+			continue
+		}
+		if currResult == revocation.Revoked {
+			break
+		}
```

Comment:
Can we add break at line 585?

Reply:
No, that would have it break out of the switch, not the outer for loop. This will be revised when combining the loops as suggested in the above comment